### PR TITLE
Bug 1948533: Add nodeip-configuration.service for oVirt

### PR DESCRIPTION
**Pull request by Gal-Zaidman**
**- What I did**

This change adds the nodeip-configuration.service to the oVirt platform.
On 4.7 due to the unification of the unit files of each on prem platform to the "on-prem" directory oVirt got the "nodeip-configuration.service" file but on 4.6 release we are missing it which can cause hard to debug problems to customers.

This change also adds fixes for BZ#1940939 https://github.com/openshift/machine-config-operator/pull/2470

** I was not able to reproduce the issue on BZ#1948533 but this change seems relevant and according to the comments on BZ#1948533 will fix the issues

**- How to verify it**
Install a 4.6 ovirt cluster

**- Description for the changelog**

### CODE REVIEW

**Comment by Gal-Zaidman** [comment]
/hold

For tests

**Comment by Gal-Zaidman** [comment]
/unhold

oVirt job passed

File: templates/common/ovirt/units/nodeip-configuration.service.yaml
Comment (patrickdillon) [inline templates/common/ovirt/units/nodeip-configuration.service.yaml:2]: Do you want to disable this service in the case of a UPI install? That is what we do for vSphere, but not sure if applicable for your case: https://github.com/openshift/machine-config-operator/blob/release-4.6/templates/common/vsphere/units/nodeip-configuration.service.yaml

**Comment by patrickdillon** [comment]
Fixes #2529

**Review by yuqi-zhang — APPROVED** [review]
Generally LGTM pending Patrick's comment. Will also let him do the final lgtm when ready

File: templates/common/ovirt/units/nodeip-configuration.service.yaml
Comment (Gal-Zaidman) [reply templates/common/ovirt/units/nodeip-configuration.service.yaml]: Hi thanks for the review,
I added the enabled logic but removed the ifs from the "contents" since I think that if the service is disabled it doesn't matter

File: templates/common/ovirt/units/nodeip-configuration.service.yaml
Comment (patrickdillon) [inline templates/common/ovirt/units/nodeip-configuration.service.yaml:48]: Does this attribute exist in UPI? We had the conditionals to protect against an exception in UPi where this value does not exist.

Comment (Gal-Zaidman) [reply templates/common/ovirt/units/nodeip-configuration.service.yaml:48]: Nope, it wouldn't exist, but I thought that if the service is disabled it doesn't matter.
That is not the case?

Comment (patrickdillon) [reply templates/common/ovirt/units/nodeip-configuration.service.yaml:48]: The template is written regardless, so without the condition the mco will throw an exception trying to access the value while writing the template.

Comment (Gal-Zaidman) [reply templates/common/ovirt/units/nodeip-configuration.service.yaml:53]: thanks for the explanation, fixed

**Pull request merged**
